### DISCUSSION TITLE
Wrap response object in static object result: { ... }

### DISCUSSION THREAD

**Discussion by r0otx** [fastapi/fastapi#9889] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class ResultModel(BaseModel):
    result: dict

@app.middleware("http")
async def wrap_response(request: Request, call_next):
    response = await call_next(request)
    data = await response.json()
    result = ResultModel(result=data)
    return JSONResponse(content=result.dict())
```

### Description

I'm try make middleware with this code. But was see error: response don't contain json method. How I can change all responses in my endpoints without lost pydantic functionality and I can get all responses wrapped in object "result": {...}?

### Operating System

Linux

### FastAPI Version

0.95.2

### Python Version

3.11

**Comment by jgould22** [comment, marked as answer]
So a few things.

In your example you are trying to treat a response like a request. 

This is the type you are getting back in your response https://github.com/encode/starlette/blob/5771a78c14d577bd1743ed2e948c91589dd30e20/starlette/responses.py#L39C7-L39C7 as you can see it does not have a JSON method. 

The you probably want the content attribute which contains the data you want.

That being said there are some implications if you go about this the way you are trying to here.

- This will probably break the documentation.
- Might only work on JSON response
- Is recreating the JSON response so anything set on the response (like headers or cookies or status codes) will get tossed out.

Another way to get a similar response is to define a Generic Model and set it as your response model in the controller. You can see an [example here](https://stackoverflow.com/questions/75333446/fastapi-create-a-generic-response-model-that-would-suit-requirements). However that will not allow you to do what I believe you are trying to achieve which is to just return some model in the controller and have it wrapped in that result.
